app/root.tsx

```diff
@@ -5,15 +5,21 @@ import { TopNav } from '~/components/layout/top-nav';
 import { Links, LiveReload, Meta, Outlet, Scripts, ScrollRestoration } from '@remix-run/react';
 import Footer from './components/layout/footer';
 import { getCurrentUser } from '~/modules/session/session.server';
+import { db } from './modules/database/db.server';
 
 export const links: LinksFunction = () => [{ rel: 'stylesheet', href: stylesheet }];
 
 export async function loader({ request }: LoaderArgs) {
+  const events = await db.event.findMany({ include: { group: true }, take: 24 });
+  const groups = await db.group.findMany({ take: 24 });
```

Comment:
We can use `Promise.all` here to avoid a request waterfall! Currently, we have to wait for the events to be queried before we start querying the groups. We can start both queries in parallel like so:

```suggestion
  const eventsPromise = db.event.findMany({ include: { group: true }, take: 24 });
  const groupsPromise = db.group.findMany({ take: 24 });
  const [events, groups] = await Promise.all([eventsPromise, groupsPromise]);
```

Reply:
Thanks for the great suggestion!